lua/ccc/picker/custom_entries.lua

```diff
@@ -0,0 +1,78 @@
+local config = require("ccc.config")
+local utils = require("ccc.utils")
+local parse = require("ccc.utils.parse")
+
+---@class CustomEntries: ColorPicker
+---@field rgb { [string]: integer[] }
+---@field re { match_str: function } vim.regex instance
+---@field min_length integer
+---@field color_table { [string]: string }
+---@field exclude_pattern_option string[]|string
+local CustomEntries = {}
+
+---@param color_table { [string]: string }
+---@return CustomEntries
+CustomEntries.new = function(color_table)
+  return setmetatable(
+    { color_table = color_table, rgb = {}, pattern = {}, exclude_pattern = {} },
+    { __index = CustomEntries }
+  )
+end
+
+function CustomEntries:init()
+  if self.re then
+    return
+  end
+  ---@type { plain: string, vim: string }[]
+  local patterns = {}
+  for name, rgb in pairs(self.color_table) do
+    local r, g, b = rgb:match("(%x%x)(%x%x)(%x%x)")
+    if r then
+      local escaped = name:gsub([[\]], "%0%0")
+      table.insert(patterns, { plain = name, vim = escaped })
+      self.rgb[name] = { parse.hex(r), parse.hex(g), parse.hex(b) }
+    end
+  end
+  -- Sort names to match the longest word at first.
+  table.sort(patterns, function(a, b)
+    return #a.plain > #b.plain
+  end)
+  self.min_length = #patterns[#patterns].plain
+  local vim_patterns = vim.tbl_map(function(v)
+    return v.vim
+  end, patterns)
+  local pattern = [[\V\<\(]] .. table.concat(vim_patterns, [[\|]]) .. [[\)\>]]
+  self.exclude_pattern_option = config.get("exclude_pattern").custom_entries
```

Comment:
I thought about it for a while and decided to do away with exclude_pattern altogether. The reason is that the main purpose of it can be solved with vim's regexp `\<\>`, and a simpler design is preferable. So please remove this part.
Thank you for making me aware of `\<\>`.

Reply:
I see. So you think you will remove the feature after this PR is merged? Then I remove this ahead of that.

Reply:
Fixed by f16463a

Reply:
Ah, I have noticed `exclude_pattern` is already removed in main.